R/bootstrap.R

```diff
@@ -7,6 +7,19 @@
   if (is.null(x)) y else x
 }
 
+catf <- function(fmt, ..., appendLF = TRUE) {
+
+  quiet <- getOption("renv.bootstrap.quiet", default = FALSE)
+  if (quiet)
+    return(invisible())
+
+  msg <- sprintf(fmt, ...)
+  cat(msg, file = stdout(), sep = if (appendLF) "\n" else "")
```

Comment:
TIL that R does something odd if `sep` contains a newline; in particular, it writes a trailing newline if `sep` _contains_ a newline. I guess I'm taking advantage of that here, but it's definitely a little odd.

`writeLines()` is more consistent in this regard.